contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -280,10 +288,126 @@ namespace eosiosystem {
+    };
+
+    struct [[eosio::table, eosio::contract("eosio.system")]] proposal {
+        name proposer;        // proposer
+        uint64_t id;
+        name committee;       // committee
+        string category;              // category
+        uint16_t subcategory;         // subcategory
+        string title;                 // title
+        string summary;               // summary
+        string project_img_url;       // project image or video url
+        string description;           // overview
+        string roadmap;               // roadmap
+        uint64_t duration;            // duration
+        vector<string> members;       // linkedin
+        asset funding_goal;           // amount of EOS
+        double total_votes;         // total votes
+        uint8_t status;               // status
+        time_point_sec vote_start_time;     // time when voting starts (seconds)
+        time_point_sec fund_start_time;     // time when funding starts (seconds)
+        uint32_t iteration_of_funding; // current number of iterations
+        uint32_t total_iterations; // total number of iterations
+        uint64_t primary_key() const { return proposer.value; }
+        uint64_t by_id() const { return id; }
+        double   by_votes()const    { return total_votes;  }
+        EOSLIB_SERIALIZE( proposal, (proposer)(id)(committee)(category)(subcategory)(title)(summary)(project_img_url)(description)(roadmap)(duration)(members)(funding_goal)
+                (total_votes)(status)(vote_start_time)(fund_start_time)(iteration_of_funding)(total_iterations) )
+    };
+
+    struct [[eosio::table, eosio::contract("eosio.system")]] committee {
+        name committeeman;
+        string category;
+        bool is_oversight;
+        uint64_t primary_key() const { return committeeman.value; }
+        EOSLIB_SERIALIZE( committee, (committeeman)(category)(is_oversight) );
+    };
+
+    struct [[eosio::table, eosio::contract("eosio.system")]] reviewer {
+        name account;
+        name committee;
+        string first_name;
+        string last_name;
+        uint64_t primary_key() const { return account.value; }
+        EOSLIB_SERIALIZE( reviewer, (account)(committee)(first_name)(last_name) )
+    };
+
+    struct [[eosio::table("wpsglobal"), eosio::contract("eosio.system")]] wpsenv {
+        uint32_t total_voting_percent = 5;           // 5%
+        uint32_t duration_of_voting = 30;            // voting duration (days)
+        uint32_t max_duration_of_funding = 500;      // funding duration (days)
+        uint32_t total_iteration_of_funding = 6;     //
+        uint64_t primary_key() const { return 0; }
+        EOSLIB_SERIALIZE( wpsenv, (total_voting_percent)(duration_of_voting)(max_duration_of_funding)(total_iteration_of_funding) )
+    };
+
+    /**
+    * Proposers table
+    *
+    * @details The proposers table stores all WPS proposers' information
+    */
+    typedef eosio::multi_index<"proposers"_n, proposer> proposer_table;
```

Comment:
Seems to be better to use `using` instead of `typedef`:
`using proposer_table = eosio::multi_index<"proposers"_n, proposer>;`

Reply:
I'm going to have to advise a hard no on changing this one.
Doing a `typedef` for multi_index tables is common and best practices for EOSIO smart contracts. 
In fact, the current version of the WAX mainnet system contracts (which are currently deployed on WAX mainnet) use `typedef`s on the multi_index tables: https://github.com/worldwide-asset-exchange/wax-system-contracts/blob/develop/contracts/eosio.system/include/eosio.system/eosio.system.hpp#L125
All of Block.one's original smart contracts do the same:
https://github.com/EOSIO/eosio.contracts/blob/master/contracts/eosio.token/include/eosio.token/eosio.token.hpp#L171

Reply:
Ok, I suppose it is better if we keep consistency with the rest of the world.